tests/unit/moduleapi/usercall.tcl

```diff
@@ -0,0 +1,52 @@
+set testmodule [file normalize tests/modules/usercall.so]
+
+set test_script_set "#!lua
+redis.call('set','x',1)
+return 1"
+
+set test_script_get "#!lua
+redis.call('get','x')
+return 1"
+
+start_server {tags {"modules usercall"}} {
+    r module load $testmodule
+
+    test {test module check regular redis command without user/acl} {
+        assert_equal [r usercall.call_without_acl set x 5] OK
+    }
+
+    test {test module check regular redis command with user/acl} {
+        assert_equal [r set x 5] OK
+
+        assert_equal [r usercall.reset_user] OK
+        assert_equal [r usercall.add_to_acl "~* &* +@all -set"] OK
+        # off because module user / default value
+        assert_equal [r usercall.get_acl] "off ~* &* +@all -set"
+
+        catch {r usercall.call_with_acl set x 10} e
+        assert_match {*ERR acl verification failed*} $e
+
+        assert_equal [r usercall.call_with_acl get x] 5
+    }
+
+    test {test module check eval script with user/acl} {
```

Comment:
how about

call_without_user (regular rm_call)
call_with_user (set user, regular rm_call)
call_with_user_and_acl (set user and rm_call with 'C' flag)

Reply:
I think I addressed this in latest push

Reply:
i still think that considering that call_with_user and call_with_user_and_acl are identical except the flag
we better just let the tcl code pass the flags, and then it can also use other flags.
so we'll have just two commands, one uses a user and one that doesn't.

p.s. where you say "with acl", i think it should say "with acl checks", to make a difference between setting up ACL user, and asking RM_Call to do the checks.

Reply:
I'm unconvinced that they are equivalent.

call_with_user doesn't check ACL, just sets the user on the context, while ACL actually calls it with ACL checking.

i.e. we have 3 ways to do the call

1) base rm_call (should be effectively same as before)
2) with user - but as not specifying ACL, nothing should change (should behave the same as regular rm_call)
3) with acl - now, this is where the difference is.

Reply:
my point was that the two functions are identical except for the flags that are passed, so if we pass them from outside we have:
1. more flexibility to the caller (can add other flags)
2. all the context about the test is in one place. i.e. the reader needs to read less C (not that many functions with a ton of code in each), and when he gets to read the TCL code, he can still remember what each C command does.

when i read these module tests code, i often have a problem where i need to run back and forth between the tcl code and the c code on each line of tcl code that i'm reading.
if the C code is simple, and i can clearly understand what each module command does, then it's sufficient to read just the TCL code (assuming i know the RM_Call feature and meaning of the flags, since if i don't know that, i need to also switch between a TCL file and 2 C files).

maybe if we take if further, we can even have just one rm_call wrapper command, and control the selected user using a specific command that does just that.

anyway, i won't insist on that.. just a suggestion for something that can maybe make the tests more readable.

Reply:
ok, so if I understand correctly, it would be a call_with_user and call_with_user_flags like in misc.c, where right now it just be used with the acl flag of 'C'